### PR TITLE
fix(requests): improve createRequestDataSource typing

### PR DESCRIPTION
**Pull request by EricPoul**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/elf/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: [72](https://github.com/ngneat/elf/issues/72)

## What is the new behavior?

Types work correctly and match the following:
- If the user provides requestKey the returned value should be DataSource.
- If the user didn't provide requestKey the returned value should be DynamicKeyDataSource.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by EricPoul** [comment]
@NetanelBasal I haven't changed or added tests since changes were only about types issues but if you think I should have done then I will do

**Comment by NetanelBasal** [comment]
Yes, it'll be great. I didn't catch this bug because I didn't use the generic in my spec. Would you please add a simple spec with both types to the `createRequestDataSource` spec file?

File: packages/requests/src/lib/request-data-source.ts [hunk 1]
```diff
@@ -58,31 +58,73 @@ type DynamicKeyDataSource<S, DataKey extends string, Data, Error> = {
...
+  | DynamicKeyDataSource<S, DataKey, Data, Error>
+  | DataSource<S, DataKey, Data, Error> {
+  const {
+    data$,
+    store,
+    dataKey,
+    requestStatusOptions,
+    requestKey,
+    idleAsPending = false,
+  } = 'requestKey' in params ? params : { ...params, requestKey: undefined };
```
Comment (NetanelBasal) [inline packages/requests/src/lib/request-data-source.ts:126]: Can you please change it to `Reflect.has(params, 'requestKey')`

Comment (NetanelBasal) [reply packages/requests/src/lib/request-data-source.ts:126]: np

**Pull request merged**